components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/cmd/AddCmd.java

```diff
@@ -430,11 +435,62 @@ else if(typeAndProjectName[0].equals("route")){
                     GatewayCmdUtils.saveConfig(newConfig, toolkitConfigPath);
                 }
             }
+        }//todo: implement add route command
+        else {
+            if (resource_id.isEmpty()) {
+                if ((endpoint = promptForTextInput("Enter Resource ID: "))
+                        .trim().isEmpty()) {
+                    throw GatewayCmdUtils.createUsageException("Micro gateway add route failed: " +
+                            "resource_id is not provided");
+                }
+            }
+
+            String endpointConfigString;
+            if (StringUtils.isEmpty(endpointConfig)) {
+                if (StringUtils.isEmpty(endpoint)) {
```

Comment:
Can you add this as in bellow.
if (StringUtils.isEmpty(endpointConfig) && StringUtils.isEmpty(endpoint))

Reply:
Then we will have to bring another else if as (StringUtils.isEmpty(endpointConfig) && !StringUtils.isEmpty(endpoint)). And further, it is a chained code segment. If it is implemented as suggested there would be redundant code as well.

Reply:
yes agreed.